pkg/apis/minio.min.io/v2/helper.go

```diff
@@ -1172,6 +1172,13 @@ func (t *Tenant) GetDomainHosts() []string {
 			if err != nil {
 				continue
 			}
+			// No slash after scheme (see https://pkg.go.dev/net/url#URL)
+			if u.Host == "" {
+				u, err = url.Parse("https://" + d)
+				if err != nil {
+					continue
+				}
+			}
```

Comment:
This looks like a hack @allanrogerr we should not do it this way look at how we validate this in MinIO - especially url.Parse must not be used here

```go
        domains := env.Get(config.EnvDomain, "")
        if len(domains) != 0 {
                for _, domainName := range strings.Split(domains, config.ValueSeparator) {
                        if _, ok := dns2.IsDomainName(domainName); !ok {
                                // fail here...
                        }
                        globalDomainNames = append(globalDomainNames, domainName)
                }
                sort.Strings(globalDomainNames)
                lcpSuf := lcpSuffix(globalDomainNames)
                for _, domainName := range globalDomainNames {
                        if domainName == lcpSuf && len(globalDomainNames) > 1 {
                                // fail here.. 
                        }
                }
        }
```

And for `lcp` use this code

```go
// Suffix returns the longest common suffix of the provided strings
func lcpSuffix(strs []string) string {
        return lcp(strs, false)
}

func lcp(strs []string, pre bool) string {
        // short-circuit empty list
        if len(strs) == 0 {
                return ""
        }
        xfix := strs[0]
        // short-circuit single-element list
        if len(strs) == 1 {
                return xfix
        }
        // compare first to rest
        for _, str := range strs[1:] {
                xfixl := len(xfix)
                strl := len(str)
                // short-circuit empty strings
                if xfixl == 0 || strl == 0 {
                        return ""
                }
                // maximum possible length
                maxl := xfixl
                if strl < maxl {
                        maxl = strl
                }
                // compare letters
                if pre {
                        // prefix, iterate left to right
                        for i := 0; i < maxl; i++ {
                                if xfix[i] != str[i] {
                                        xfix = xfix[:i]
                                        break
                                }
                        }
                } else {
                        // suffix, iterate right to left
                        for i := 0; i < maxl; i++ {
                                xi := xfixl - i - 1
                                si := strl - i - 1
                                if xfix[xi] != str[si] {
                                        xfix = xfix[xi+1:]
                                        break
                                }
                        }
                }
        }
        return xfix
}
```

Reply:
Modified code to suit